src/annotator/integrations/test/pdf-test.js

```diff
@@ -369,15 +373,119 @@ describe('PDFIntegration', () => {
+    });
+
+    it('does not scroll if anchor has no highlights', async () => {
       const integration = createPDFIntegration();
+      const anchor = {};
+
       await integration.scrollToAnchor(anchor);
 
-      assert.calledOnce(fakeScrollIntoView);
-      assert.calledWith(fakeScrollIntoView, highlight, sinon.match.func);
+      assert.notCalled(fakeScrollUtils.scrollElement);
+    });
+
+    /**
+     * Create an anchor whose highlight is inside a placeholder for a non-rendered
+     * PDF page.
+     */
+    function createPlaceholderHighlight() {
+      const container = document.createElement('div');
+      const placeholder = createPlaceholder(container);
+      const highlight = document.createElement('div');
+      placeholder.append(highlight);
+      return highlight;
+    }
+
+    it('waits for anchors in placeholders to be re-anchored and scrolls to final highlight', async () => {
+      const placeholderHighlight = createPlaceholderHighlight();
+      const integration = createPDFIntegration();
+      fakeScrollUtils.offsetRelativeTo
+        .withArgs(placeholderHighlight, integration.contentContainer())
+        .returns(50);
+      const annotation = { $tag: 'tag1' };
+      const anchor = { annotation, highlights: [placeholderHighlight] };
+
+      // Check that the PDF content was scrolled to the approximate position of
+      // the anchor, indicated by the placeholder.
+      const scrollDone = integration.scrollToAnchor(anchor);
+      assert.calledWith(
+        fakeScrollUtils.scrollElement,
+        integration.contentContainer(),
+        50
+      );
+
+      // Simulate a delay while rendering of the text layer for the page happens
+      // and re-anchoring completes.
+      await delay(5);
+
+      // Create a new anchor for the annotation created by re-anchoring.
+      const finalHighlight = document.createElement('div');
+      fakeScrollUtils.scrollElement.resetHistory();
+      fakeAnnotator.anchors.push({
+        annotation,
+        highlights: [finalHighlight],
+      });
+      fakeScrollUtils.offsetRelativeTo
+        .withArgs(finalHighlight, integration.contentContainer())
+        .returns(150);
+
+      await scrollDone;
+
+      // Check that we scrolled to the location of the final highlight.
+      assert.calledWith(
+        fakeScrollUtils.scrollElement,
+        integration.contentContainer(),
+        150
+      );
+    });
+
+    it('skips scrolling to final anchor if re-anchoring does not complete within timeout', async () => {
+      const highlight = createPlaceholderHighlight();
+      const integration = createPDFIntegration({ reanchoringWait: 10 });
+      const annotation = { $tag: 'tag1' };
+      const anchor = { annotation, highlights: [highlight] };
+
+      const scrollDone = integration.scrollToAnchor(anchor);
+      await delay(5); // Simulate delay in re-anchoring
+      fakeScrollUtils.scrollElement.resetHistory();
+
+      // Wait until the re-anchoring timeout expires.
```

Comment:
I understand that the `delay(5)` is not actually related to the timeout here... The timeout happens because there isn't information/elements present to allow scrolling within `10ms` (non-placeholder anchor highlights are _never_ available, `10ms` or `10000ms` from now). I presume the `10ms` value is just because it's fairly quick and lets tests run quickly...Correct?

Reply:
Right, the 10ms value is just to shorten test execution. Importantly it is less than the 2000ms timeout for tests.